third_party/terraform/utils/retry_transport.go

```diff
@@ -99,19 +98,29 @@ func (t *retryTransport) RoundTrip(req *http.Request) (resp *http.Response, resp
 	backoff := time.Millisecond * 500
 	nextBackoff := time.Millisecond * 500
 
+	// VCR depends on the original request body being consumed, so
+	// consume here. Since this won't affect the request itself,
+	// we do this before the actual Retry loop so we can consume the request Body as needed
+	// e.g. if the request couldn't be retried, we use the original request
+	if _, err := httputil.DumpRequestOut(req, true); err != nil {
```

Comment:
This will cause an error for non-retried requests I believe?

DumpRequestOut consumes the request body of `req` which we later try to send directly in some cases: `resp, respErr = t.internal.RoundTrip(req)`

Wouldn't sending that request fail if the body has already been consumed?

Reply:
no - [DumpRequestOut](https://golang.org/src/net/http/httputil/dump.go?s=2131:2196#L68) consumes the body bytes, reading it into memory, and then replaces the request body with an Reader with the same bytes, so the actual request isn't really changed. 

I am conflicted though - originally I wanted to avoid doing the dump to copy the request, since a) it creates copy in memory and b) the copy created by httputil isn't supposed to be used to actually send again, and while it might work for now, it's not clear whether that'll be true in the future. That does mean, instead of the GetBody error, we could do the copy-bytes approach if GetBody isn't given? 

I'm leaning towards keeping this as is (i.e. with the changes in this PR), since it seemed like the storage/googleapi libraries purposely doesn't set GetBody if the request isn't retryable.

Reply:
Should we at least dump the request after we copy/send it so that the original body is sent rather than the copy?

 Yeah, I would say that since we are seeing this error come up with non-retryable requests (uploading large files) that we should assume that if GetBody isn't set we shouldn't retry.